studio/schemas/schemaTypes/slug.ts

```diff
@@ -19,26 +19,42 @@ async function isSlugUniqueAcrossAllDocuments(
   return result;
 }
 
+const SLUG_MAX_LENGTH = 200;
+
 function createSlugField(source: string) {
   return defineField({
     type: "slug",
     name: "slug",
     title: "URL Path (slug)",
     description:
-      "Enter a unique URL path for the page. This path will be used in the website's address bar. A URL path, also known as a slug, is a URL-friendly version of the page title, used to create a human-readable and search engine optimized URL for the content.",
+      "Enter a unique URL path for the page. This path will be used in the website's address bar. A URL path, also known as a slug, is a URL-friendly version of the page title, used to create a human-readable and search engine optimized URL for the content. Legal characters include latin letters, digits, hyphen (-), underscore (_), full stop (.) and tilde (~)",
     options: {
       source,
-      maxLength: 200,
+      maxLength: SLUG_MAX_LENGTH,
       slugify: (input) =>
-        input.toLowerCase().replace(/\s+/g, "-").slice(0, 200),
+        input
+          .toLowerCase()
+          // replace æøå according to https://sprakradet.no/spraksporsmal-og-svar/ae-o-og-a-i-internasjonal-sammenheng/
+          .replace(/[æ,å]/g, "a")
+          .replace(/ø/g, "o")
+          .replace(/[^a-zA-Z0-9-_.~\s]/g, "") // remove non-whitespace URL-unsafe chars (section 2.3 in https://www.ietf.org/rfc/rfc3986.txt)
+          .trim()
+          .replace(/\s+/g, "-")
+          .slice(0, SLUG_MAX_LENGTH),
       isUnique: isSlugUniqueAcrossAllDocuments,
     },
-    validation: (Rule) => Rule.required(),
+    validation: (Rule) =>
+      Rule.required().custom((value) => {
+        if (value?.current === undefined) return true;
+        return (
+          encodeURIComponent(value.current) === value.current ||
+          "Slug can only consist of latin letters, digits, hyphen (-), underscore (_), full stop (.) and tilde (~)"
+        );
+      }),
   });
 }
 
 const pageSlug = createSlugField("page");
-const blogSlug = createSlugField("basicTitle");
-const legalSlug = createSlugField("basicTitle");
+const pageTitleSlug = createSlugField("basicTitle");
 
-export { pageSlug, blogSlug, legalSlug };
+export { pageSlug, pageTitleSlug };
```

Comment:
just be second guessing myself here.

do you agree or disagree that this makes sense?

my thought was that if it's pagetitle, it explicitly says it's the slug generated from the page title and to be used within page schemas, and if it's pageSlug it's to be used in pages without a title, but a page name because these are names used for sanity studio identification and for slug as it can be different from the title used on screen.. 

which again makes me think that we shouldn't be using the title in salaryAndBenefits to generate the slug as we want the slug to be `/kalkulator` but the title might be "Hos variant har vi drit kule goder" 💡  Ofc the user can manually set the slug by not generating from title.. 🤔

Reply:
for page we needed to generate because title is set in section.hero.. 💡 so perhaps it's ok that we're using title anyway in salary and benefits!

Reply:
But perhaps the value should then be 

`titleSlug` and not `pageTitleSlug` like original?

Reply:
I agree that `titleSlug` is better (or even `basicTitleSlug`). I just kept the other exports to avoid changing other parts of the code, but it retrospect that might just make it more confusing. 

We don't have anything other than title to base the slug on, but we could always make the slug have some static value as source for "Generate". I think the title based slug is fine though.

Reply:
Yea, I think titleSlug or basicTitleSlug might be better, yes!

Reply:
Are we good then?

Reply:
Yes! I forgot to set "approved"